pandas/tests/io/test_clipboard.py

```diff
@@ -22,73 +23,139 @@
-        del cls.data_types, cls.data
-
-    def check_round_trip_frame(self, data_type, excel=None, sep=None,
+    def check_round_trip_frame(self, data, excel=None, sep=None,
                                encoding=None):
-        data = self.data[data_type]
         data.to_clipboard(excel=excel, sep=sep, encoding=encoding)
-        if sep is not None:
-            result = read_clipboard(sep=sep, index_col=0, encoding=encoding)
-        else:
-            result = read_clipboard(encoding=encoding)
+        result = read_clipboard(sep=sep or '\t', index_col=0,
+                                encoding=encoding)
         tm.assert_frame_equal(data, result, check_dtype=False)
 
-    def test_round_trip_frame_sep(self):
-        for dt in self.data_types:
-            self.check_round_trip_frame(dt, sep=',')
-            self.check_round_trip_frame(dt, sep=r'\s+')
-            self.check_round_trip_frame(dt, sep='|')
-
-    def test_round_trip_frame_string(self):
-        for dt in self.data_types:
-            self.check_round_trip_frame(dt, excel=False)
-
-    def test_round_trip_frame(self):
-        for dt in self.data_types:
-            self.check_round_trip_frame(dt)
+    # Test that default arguments copy as tab delimited
+    # Fails because to_clipboard defaults to space delim.
+    # Issue in #21104, Fixed in #21111
+    @pytest.mark.xfail
+    def test_round_trip_frame(self, df):
+        self.check_round_trip_frame(df)
+
+    # Test that explicit delimiters are respected
+    @pytest.mark.parametrize('sep', ['\t', ',', '|'])
+    def test_round_trip_frame_sep(self, df, sep):
+        self.check_round_trip_frame(df, sep=sep)
+
+    # Test white space separator
+    # Fails on 'delims' df because quote escapes aren't handled correctly
+    # in default c engine. Fixed in #21111 by defaulting to python engine
+    # for whitespace separator
+    @pytest.mark.xfail
+    def test_round_trip_frame_string(self, df):
+        df.to_clipboard(excel=False, sep=None)
+        result = read_clipboard()
+        assert df.to_string() == result.to_string()
+        assert df.shape == result.shape
+
+    # Two character separator is not supported in to_clipboard
+    # Test that multi-character separators are not silently passed
+    # Fails, Fixed in #21111
+    @pytest.mark.xfail
+    def test_excel_sep_warning(self):
+        with tm.assert_produces_warning():
+            gen_df('string').to_clipboard(excel=True, sep=r'\t')
+
+    # Separator is ignored when excel=False and should produce a warning
+    # Fails, Fixed in #21111
+    @pytest.mark.xfail
+    def test_copy_delim_warning(self):
+        with tm.assert_produces_warning():
+            gen_df('string').to_clipboard(excel=False, sep='\t')
+
+    # Tests that the default behavior of to_clipboard is tab
+    # delimited and excel="True"
+    # Fails because to_clipboard defaults to space delim.
+    # Issue in #21104, Fixed in #21111
+    @pytest.mark.xfail
```

Comment:
every xfail needs a reason

Reply:
Can you take a look at the latest version? Is that what you meant for reason?